aten/src/ATen/native/cuda/LogcumsumexpKernel.cu

```diff
@@ -11,24 +11,106 @@
 
 namespace at::native {
 
+// custom min and max to be used in logcumsumexp for complex arguments
+template <typename scalar_t, bool min>
+__host__ __device__ c10::complex<scalar_t> _logcumsumexp_minmax(const c10::complex<scalar_t>& x, const c10::complex<scalar_t>& y) {
```

Comment:
Actually you can revert the templating arg too, its' a bit difficult to setup this in a constexpr if statement that is clean with the all the non-constexpr conditions as well.

Reply:
Also all the else statements are unnecessary since they all have return statements in them.

Reply:
@mfkasim1 just removes extra indentation. That's why it's a nit. Don't really care either way.